First pass - intro & shortcuts (#629)
MicrosoftDocs/edge-developer: microsoft-edge/progressive-web-apps-chromium/webappmanifests.md

---

> title: Deeper OS integrations via the Web App Manifest

Reviewer: ```suggestion
title: Use the Web App Manifest to integrate your Progressive Web App into the Operating System
```

> description: The Web App Manifest governs how your app looks and behaves within the context of the Operating System.

Reviewer: Learn how to use the Web App Manifest to integrate your Progressive Web App into your Operating System.

keywords: progressive web apps, PWA, Edge, JavaScript, Windows, UWP, Microsoft Store
---

# Deeper OS integrations via the Web App Manifest

> A site’s Web App Manifest governs how it looks and behaves when installed on a device. At its most basic, the Manifest provides detail about what name and icons should be used to represent your app in system menus to the theme colors the OS should use for the title bar. The Manifest also enables you to unlock more powerful features that allow your app to behave like other native apps on the system.
> 

Reviewer: ```suggestion
A site’s Web App Manifest governs how your Progressive Web App (PWA) looks and behaves when installed on a device. At its most basic, the Manifest provides details on the name of your app, icons to use to represent your app in system menus, and the theme colors the operating system (OS) uses in the title bar. The Manifest also enables you to unlock powerful features that allow your app to behave like other native apps on the system.

```

> ## Provide quick access to key features

Reviewer: ```suggestion
## Use shortcuts to provide quick access to features
```

> Most operating systems provide some sort of quick access to key app features via a context menu connected to the app’s icon. This feature is, called "shortcuts," is enabled in Progressive Web Apps via the Web App Manifest’s `shortcuts` member. Shortcuts are defined in the manifest like this:

Reviewer: ```suggestion
Most operating systems provide quick access to key app features using shortcuts on the context menu connected to the app’s icon. To use shortcuts in your PWA, include the `shortcuts` member in your Web App Manifest. The following example shows how to define a shortcut in your web app manifest. 
```

```json
  "shortcuts": [
    {
      "name": "Play Later",
      "description": "View the list of podcasts you saved for later",
      "url": "/play-later",
      "icons": [
        {
          "src": "/icons/play-later.svg",
          "type": "image/svg+xml",
          "purpose": "any"
        }
      ]
    },
    {
      "name": "Subscriptions",

>       "description": "View the list of podcasts you listen to",

Reviewer: ```suggestion
      "description": "View the list of podcasts available in your subscription",
```

      "url": "/subscriptions?sort=desc"
    }
  ]
```

> In the context of a full Web App Manifest, adding the above snippet would enable two shortcuts on the app’s icon. The first would be labelled "Play Later" and would have a custom icon. The second would be labelled "Subscriptions" and does not have an icon associated with it. As you can see, the `icons` member is optional. The `description` member is also optional and can be used to provide additional information for accessibility purposes.

Reviewer: ```suggestion
When added to a complete Web App Manifest, adding the above snippet would enable two shortcuts on the context menu on the app’s icon. The first would be labelled "Play Later" and has a custom icon. The second would be labelled "Subscriptions" and does not have an icon because the `icons` member is optional. The `description` member is also optional and may be used to provide additional information for accessibility purposes.
```
